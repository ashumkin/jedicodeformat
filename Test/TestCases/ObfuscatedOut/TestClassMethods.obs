unit testclassmethods;interface type tfooform=class private var foovar:integer;class
var fish:integer;class function getwibble:integer;static;public class function foo
:boolean;class procedure bar;class property wibble:integer read getwibble;end;implementation
class procedure tfooform.bar;begin inc(fish);end;class function tfooform.foo:boolean
;begin result:=true;end;class function tfooform.getwibble:integer;begin result:=fish
;end;end. 