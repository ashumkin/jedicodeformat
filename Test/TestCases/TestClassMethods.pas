unit TestClassMethods;

{ AFS 15 July 2007

  This code compiles, but is not semantically meaningfull.
  It is test cases for the code-formating utility

  this unit tests class methods and properties  }

interface

type
  TFooForm = class
  private
    var foo: integer;
    class var fish: integer;
    class function GetWibble: integer; static;
  public
    class function Foo: boolean;
    class procedure Bar;

    class property Wibble: integer read GetWibble;
  end;


implementation

{ TFooForm }

class procedure TFooForm.Bar;
begin
  inc(fish);
end;

class function TFooForm.Foo: boolean;
begin
  Result := true;
end;

class function TFooForm.GetWibble: integer;
begin
  Result := fish;
end;

end.
