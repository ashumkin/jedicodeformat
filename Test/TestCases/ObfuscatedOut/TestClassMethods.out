unit testclassmethods;

interface

type
  tfooform = class
  private
  var
    foo:      integer;
    class
    var fish: integer;
    class function getwibble: integer; static;
  public
    class function foo: boolean;
    class procedure bar;
    class property wibble: integer Read getwibble;
  end;

implementation

class procedure tfooform.bar;
begin
  Inc(fish);
end;

class function tfooform.foo: boolean;
begin
  Result := True;
end;

class function tfooform.getwibble: integer;
begin
  Result := fish;
end;

end.
 